src/main.ts

```diff
@@ -437,30 +447,9 @@ function sanitizeErrorText(text: string): string {
 }
 
 function reduceSpew(message: string): string {
-    // Since this is only a warning, it tends to be reported many (i.e. thousands of) times
-    const problemString = "npm WARN tar ENOSPC: no space left on device, write\n";
-    const index = message.indexOf(problemString);
-    if (index < 0) return message;
-
-    return message.substring(0, index) + problemString + replaceAll(message.substring(index), problemString, "");
-}
-
-function replaceAll(message: string, oldStr: string, newStr: string) {
-    let result = "";
-    let index = 0;
-    while (true) {
-        const newIndex = message.indexOf(oldStr, index);
-        if (newIndex < 0) {
-            return index === 0
-                ? message
-                : result + message.substring(index);
-        }
-
-        result += message.substring(index, newIndex);
-        result += newStr;
-
-        index = newIndex + oldStr.length;
-    }
+    // These are uninteresting in general and actually problematic when there are
+    // thousands of instances of ENOSPC (which also appears as an error anyway)
+    return message.replace(/npm WARN.*\n/g, "");
```

Comment:
Is this per line? Should it be anchored to the beginning of the line (and is there some extra prefix you need if so)? Do you want `.replaceAll`?